testutils/utils.go

```diff
@@ -79,3 +81,29 @@ func ToJsonOrDie(value interface{}) []byte {
 		panic(err)
 	}
 }
+
+func WaitForAlertManagerSendGridReceiver(
+	k8sClient client.Client,
+	ctx context.Context,
+	amconfigKey client.ObjectKey,
+	emailadresses []string,
+	timeout time.Duration,
+	interval time.Duration,
+) {
+	var resIndex int
+	EventuallyWithOffset(1, func() string {
+		amconfig := &promv1a1.AlertmanagerConfig{}
+		ExpectWithOffset(2, k8sClient.Get(ctx, amconfigKey, amconfig)).Should(Succeed())
+		for i := range amconfig.Spec.Receivers {
+			reciever := &amconfig.Spec.Receivers[i]
+			if reciever.Name == "SendGrid" {
```

Comment:
Can we pass the receiver's name from the outside. this is util and as such should not assume a specific name.
This goes hand in hand with the comment above about the util name